### DISCUSSION TITLE
[bug] Large files slow to load

### DISCUSSION THREAD

**Discussion by ThreeTwoBravo** [teia-community/teia-ui#267] [Ideas]
### Describe the problem

With the allowing of bigger files sizes its frustrating not knowing for certain if your art work will load/play without buffering. People are expecting playback to be smooth and seamless these days especially when coming from outside the crypto art world. I have a current work at 1.35GB on Teia, while I very much appreciate that it was placed on the Teia IPFS cache. I have been told by people that it still buffers. I understand the issues around ipsf and its limitations but going forward it would be great if some solutions could be found otherwise it defeats the purpose of allowing large file sizes.  

### Describe the solution you'd like

I would be happy to pay a extra fee that ensures the large file would load and play without buffering. It would be great if this happened at the minting stage of the process but would be happy with any solution that works for the platform. 

### Alternatives considered

I do have a paid Pinata account that allows for the use of my own gateway. Would it be possible to point to that link in the metadata?

**Comment by melMass** [comment]
Hey @ThreeTwoBravo, I'm not sure to understand the request right. 

Video files are buffered and need a % of bytes loaded to start playing.
If you are expecting a 1.35gb file to load instantly I'm afraid that's a bit unrealistic

Being able to setup your own gateway is something we plan but that's a local/user setting not something that will reflect to other (like if we used metadata as you suggest), but I don't think you'll get better performance then our cache layer. Have you tried loading your file on your own gateway? Does it load faster than on teia?

**Comment by Zir0h** [comment]
```
Duration                                 : 1 min 20 s
Bit rate                                 : 135 Mb/s
Width                                    : 1 536 pixels
Height                                   : 2 048 pixels
Display aspect ratio                     : 0.750
Frame rate mode                          : Constant
Frame rate                               : 15.000 FPS
```
The video you shared on discord has a higher bitrate than my homeconnection, so for me (even with a "premium gateway") it would need some time to buffer.

I think the better solution would be to look into better compression for the video or making the resolution a bit smaller. Otherwise only people on fiber connections will be able to play it instantly.

**Comment by ThreeTwoBravo** [comment]
@melMass @Zir0h 
Thanks for the reply's. Ok cool, I'll start using lower resolutions and adjust the bit rates. It totally slipped my mind about user's internet speeds as well. 

Yea the cache layer for me works really well even on mobile which was a really nice, but I do have decent fiber. Using my own gateway it plays straight away but the difference between that and the cache layer is minimal for me. It was faster for someone I sent my gateway way link to tho. Pinata also have a streaming function for their gateways but at the moment its capped at 1080p so that's not really a option yet. Still I really appreciate the fact teia is allowing bigger file sizes. I'm pretty much using this to test what's possible. Is there anything is the Teia doc's advising best practice for minting large files(sorry I haven't looked yet)?

Also are all new mints queued to get placed on the cache layer?

**Comment by Zir0h** [comment]
I don't think there are docs on the bigger upload functions yet.

Yes, everything eventually makes it to the cache :) sometimes it takes a while though, nftstorage peering isn't great :/
